sunpy/net/jsoc/jsoc.py

```diff
@@ -489,165 +497,131 @@ def _process_time(self, time):
-                   wavelength=wavelength, segment=segment)
-
-        return dataset
+        # Extract and format primekeys
+        pkstr = ''
+        c = drms.Client()
+        pkeys = c.pkeys(series)
 
-    def _make_query_payload(self, start_time, end_time, series, notify=None,
-                            protocol='FITS', compression='rice', **kwargs):
-        """
-        Build the POST payload for the query parameters
-        """
+        for pkey in pkeys:
+            if pkey in ['T_OBS', 'T_REC', 'T_START']:
+                pkstr += '[{start}-{end}{sample}]'.format(
+                    start=start_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
+                    end=end_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
+                    sample=sample)
 
-        if protocol.upper() == 'FITS' and compression and compression.lower() == 'rice':
-            jprotocol = 'FITS,compress Rice'
-        elif protocol.upper() == 'FITS':
-            jprotocol = 'FITS, **NONE**'
-        else:
-            jprotocol = protocol
-
-        if not notify:
-            raise ValueError("JSOC queries now require a valid email address "
-                             "before they will be accepted by the server")
-
-        dataset = self._make_recordset(start_time, end_time, series, **kwargs)
-        kwargs.pop('wavelength', None)
-        kwargs.pop('sample', None)
-
-        # Build full POST payload
-        payload = {'ds': dataset,
-                   'format': 'json',
-                   'method': 'url',
-                   'notify': notify,
-                   'op': 'exp_request',
-                   'process': 'n=0|no_op',
-                   'protocol': jprotocol,
-                   'requestor': 'none',
-                   'filenamefmt': '{0}.{{T_REC:A}}.{{CAMERA}}.{{segment}}'.format(series)}
-
-        payload.update(kwargs)
-        return payload
-
-    def _send_jsoc_request(self, start_time, end_time, series, notify=None,
-                           protocol='FITS', compression='rice', **kwargs):
-        """
-        Request that JSOC stages data for download
-
-        This routine puts in a POST request to JSOC
-        """
+            elif pkey == 'WAVELNTH' and wavelength is not '':
+                if isinstance(wavelength, list):
+                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
+                    wavelength = str(wavelength)
+                else:
+                    wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
+                pkstr += wavelength
 
-        payload = self._make_query_payload(start_time, end_time, series,
-                                           notify=notify, protocol=protocol,
-                                           compression=compression, **kwargs)
+            elif len(primekeys) > 0:
+                pkstr += '[{0}]'.format(primekeys.pop(pkey, ''))
 
-        r = requests.post(JSOC_EXPORT_URL, data=payload)
+            else:
```

Comment:
Yes, all primekeys are handled here. 
Line 537 does it `pkstr += '[{0}]'.format(primekeys.pop(pkey, ''))`